### ISSUE TITLE
keywords must be strings

### ISSUE THREAD

**Issue by emmanuelito** [skozlovf/flask-json#11]
Python's JSON encoder allow integer as keys (and convert them as strings), but not flask-json.

```
>>> d = { 1 : "A" }
>>> json.dumps(d)
'{"1": "A"}'

>>> jsonify(d)
<Response 15 bytes [200 OK]>

>>> json_response(data_=d)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/opt/scodoc/venv/lib/python3.9/site-packages/flask_json.py", line 145, in json_response
    response = jsonify(data_) if data_ is not None else jsonify(**kwargs)
TypeError: keywords must be strings
```

This is a problem when one wants to replace Flask's default JSON encoder by FlaskJSON.

**Comment by skozlovf** [comment]
@emmanuelito Sorry for the long response.
This happens because `json_response()` passes `data_` as keyword arguments to the `flask.jsonify()`.
But you don't need to use `json_response()` if you want to use FlaskJSON encoder - it's auto used by the `flask.jsonify()` if you initialized `FlaskJSON`. So this should work:

```python

from flask import Flask
from flask import jsonify

app = Flask('xxx')
flask_json.FlaskJSON(app)
app.config['JSON_USE_ENCODE_METHODS'] = True

class MyJsonItem(object):
    def __json__(self):
        return '<__json__>'
    def for_json(self):
        return '<for_json>'

with app.test_request_context('/fake'):
    r = jsonify(MyJsonItem())
assert r.json == '<__json__>'
```

**Comment by emmanuelito** [comment]
@skozlovf thank you ! I understand now the source of the problem.
I was in fact using `@as_json`. 
The point is that flask-json is not a drop-in replacement for regular Flask's `jsonify`, which can be annoying (maybe this detail should be stated in the doc?).

**Comment by skozlovf** [comment]
@emmanuelito 
> maybe this detail should be stated in the doc?

I'll think about it. I planned to update internals to follow new Flask 2.3 API and will improve stuff to handle this case. Let's keep the issue open.
Thanks.

**Comment by skozlovf** [comment]
Fixed/improved in `0.4.0`.
